Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Components/Pages/BulkPagesController.cs

```diff
@@ -111,6 +117,60 @@ protected override Func<IBulkPagesController> GetFactory()
             return () => new BulkPagesController();
         }
 
+        /// <summary>
+        /// Checks whether there is a duplicate tab exists.
+        /// </summary>
+        /// <param name="pageIndex">Page index to check the duplicate for.</param>
+        /// <param name="pages">Page list to check the duplicates from.</param>
+        /// <returns>True in case there is a tab with the same name and path and with the index of greater than the first occurence; false otherwise.</returns>
+        private static bool DuplicateExists(int pageIndex, string[] pages)
+        {
+            var pageName = pages[pageIndex];
+            var pageDirectory = GetPageDirectory(pageIndex, pages);
+
+            for (var index = 0; index < pageIndex; index++)
+            {
+                var loopedDirectory = GetPageDirectory(index, pages);
+                if (loopedDirectory != pageDirectory)
+                {
+                    continue;
+                }
+                else if (pageName == pages[index])
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private static string GetPageDirectory(int pageIndex, string[] pages)
+        {
+            var fullPath = string.Empty;
+            var pageLevel = GetPageLevel(pages[pageIndex]);
+            var currentLevel = pageLevel;
+            for (var index = pageIndex; index > -1; index--)
+            {
+                var currentPage = pages[index];
+                var level = GetPageLevel(currentPage);
+                if (level < currentLevel)
+                {
+                    currentLevel = level;
+                    fullPath = $"{currentPage}>{fullPath}";
+                }
+            }
+
+            return fullPath;
+        }
+
+        private static int GetPageLevel(string pageName)
+        {
+            return pageName
+                .Replace(" ", string.Empty)
```

Comment:
Should this use `TrimStart`?

Reply:
I did it on purpose to cover the cases like "> >"(with space) instead of ">>"